scaleway/resource_domain_record_test.go

```diff
@@ -0,0 +1,606 @@
+		regexp.MustCompile(`.*dedibox.*`),
+	}
+	testDomain = ""
+)
+
+func init() {
+	// retrieve the first domain is found
+	scwClient, err := sharedClientForZone(scw.Zone(""))
+	if err != nil {
+		l.Errorf("error when retrieve scwClient: %s", err)
+		return
+	}
+
+	domainRegistrarAPI := domain.NewRegistrarAPI(scwClient)
+	listDomains, err := domainRegistrarAPI.ListDomains(&domain.RegistrarAPIListDomainsRequest{
+		Status:     domain.DomainStatusActive,
+		OrderBy:    domain.ListDomainsRequestOrderByDomainAsc,
+		IsExternal: scw.BoolPtr(true),
+	})
+	if err != nil {
+		l.Errorf("error listing domains: %s", err)
+		return
+	}
+
+	for _, domain := range listDomains.Domains {
+		// skipping reserved domains
+		isReserved := false
+		for _, reservedDomain := range reservedDomains {
+			if reservedDomain.MatchString(domain.Domain) {
+				isReserved = true
+				break
+			}
+		}
+
+		if !isReserved {
+			testDomain = domain.Domain
+			break
+		}
+	}
+
+	if testDomain == "" {
+		l.Errorf("your account doesn't have any domain to test with")
+		return
+	}
+
+	l.Infof("start domain record test with domain: %s", testDomain)
+}
+
+func TestAccScalewayDomainRecord_Basic(t *testing.T) {
+	tt := NewTestTools(t)
+	// Please note: don't use `defer tt.Cleanup()` to avoid store personnal domain in cassette testdata
+
+	uuid, err := uuid.GenerateUUID()
+	if err != nil {
+		l.Errorf("error when generate UUID: %s", err)
+		return
+	}
+	testDNSZone := fmt.Sprintf("%s.%s", uuid, testDomain)
+	l.Debugf("TestAccScalewayDomainRecord_Basic: test dns zone: %s", testDNSZone)
+
+	name := "tf"
+	recordType := "A"
+	data := "127.0.0.1"
+	dataUpdated := "127.0.0.2"
+	ttl := 3600
+	ttlUpdated := 43200
+	priority := 0
+	priorityUpdated := 10
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckScalewayDomainRecordDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+					resource "scaleway_domain_record" "tf_A" {
+						dns_zone = "%s"
+						name     = "%s"
+						type     = "%s"
+						data     = "%s"
+						ttl      = %d
+						priority = %d
+					}
+				`, testDNSZone, name, recordType, data, ttl, priority),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayDomainRecordExists(tt, "scaleway_domain_record.tf_A"),
+					resource.TestCheckResourceAttr("scaleway_domain_record.tf_A", "dns_zone", testDNSZone),
+					resource.TestCheckResourceAttr("scaleway_domain_record.tf_A", "name", name),
+					resource.TestCheckResourceAttr("scaleway_domain_record.tf_A", "type", recordType),
+					resource.TestCheckResourceAttr("scaleway_domain_record.tf_A", "data", data),
+					resource.TestCheckResourceAttr("scaleway_domain_record.tf_A", "ttl", fmt.Sprint(ttl)),
+					resource.TestCheckResourceAttr("scaleway_domain_record.tf_A", "priority", fmt.Sprint(priority)),
+					testCheckResourceAttrUUID("scaleway_domain_record.tf_A", "id"),
```

Comment:
we also need to check that the record is correct on the API side also. Otherwise, we only test that terraform persist it but not whether the correct value is present on the backend